SwiftSampleProject/SwiftyVokoderTests/ManagedObjectContextTests.swift

```diff
@@ -0,0 +1,156 @@
+        XCTAssert(countOfStations > 0)
+
+        self.manager.deleteAllObjectsOfClass(Station.self, context: nil)
+        
+        self.expectationForNotification(NSManagedObjectContextDidSaveNotification,
+            object: self.manager.managedObjectContext) { _ in
+                
+                let newCountOfStations = self.manager.countForClass(Station.self)
+                XCTAssertEqual(newCountOfStations, 0)
+                XCTAssertNotEqual(countOfStations, newCountOfStations)
+                
+                return true
+        }
+        
+        guard let rootContext = self.manager.managedObjectContext.parentContext else {
+            XCTFail("Expecting the main context to have a parent context")
+            return
+        }
+        self.expectationForNotification(NSManagedObjectContextDidSaveNotification,
+            object: rootContext) { _ in
+                
+                let newCountOfStations = self.manager.countForClass(Station.self, forContext: rootContext)
+                XCTAssertEqual(newCountOfStations, 0)
+                XCTAssertNotEqual(countOfStations, newCountOfStations)
+                
+                return true
+        }
+    
+        self.manager.saveMainContext()
+        
+        self.waitForExpectationsWithTimeout(10, handler: nil)
+    }
+    
+    func testSaveAndMergeWithMainContextSavesGrandChildren() {
+        let childContext = self.manager.temporaryContext()
+        let grandChildContext = NSManagedObjectContext(concurrencyType: .PrivateQueueConcurrencyType)
+        grandChildContext.parentContext = childContext
+        
+        let countOfStations = self.manager.countForClass(Station.self)
+        XCTAssert(countOfStations > 0)
+        
+        grandChildContext.performBlockAndWait {
+            self.manager.deleteAllObjectsOfClass(Station.self, context: grandChildContext)
+        }
+        
+        self.manager.saveAndMergeWithMainContextAndWait(grandChildContext)
+        
+        let newCountOfStations = self.manager.countForClass(Station.self)
+        XCTAssertEqual(newCountOfStations, 0)
+        XCTAssertNotEqual(countOfStations, newCountOfStations)
+    }
+    
+    func testUnsavedMainContextChangesGetPassedToTempContexts() {
+        let countOfStations = self.manager.countForClass(Station.self)
+        XCTAssert(countOfStations > 0)
+
+        let childContextBeforeChanges = self.manager.temporaryContext()
+        self.manager.deleteAllObjectsOfClass(Station.self, context: nil)
+        let childContextAfterChanges = self.manager.temporaryContext()
+
+        let childCountOfStations = self.manager.countForClass(Station.self, forContext: childContextBeforeChanges)
+        XCTAssertNotEqual(countOfStations, childCountOfStations)
+        XCTAssertEqual(childCountOfStations, 0)
+        
+        let newChildCountOfStations = self.manager.countForClass(Station.self, forContext: childContextAfterChanges)
```

Comment:
[It definitely applies to methods](https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/MemoryMgmt/Articles/mmRules.html). There's no mention of variable naming in that doc, though I have a sense memory of being smacked on this in code review a while back for variable naming as well.

tl;dr You appear to be correct, but I feel like it still looks weird if you're using `new` without actually creating something.